File: Hackerank/Days 13/main.js
'use strict';

class Book {
    constructor(title, author) {
        if (this.constructor === Book) {
            throw new TypeError("Do not attempt to directly instantiate an abstract class.");
        }

        this.title = title;
        this.author = author;
    }

    display() {
        throw new Error("Method 'display()' must be implemented.");
    }
}

class MyBook extends Book {

    constructor(title, author, price) {
        super(title, author);
        this.price = price;
    }

    display() {
        console.log("Title: " + this.title);
        console.log("Author: " + this.author);
        console.log("Price: " + this.price);
    }
}

const title = "The Alchemist";
const author = "Paulo Coelho";
const price = 248;

const book = new MyBook(title, author, price);
book.display();

File: Hackerank/Days 13/main.cs
using System;

abstract class Book
{
    protected string title;
    protected string author;

    public Book(string title, string author)
    {
        this.title = title;
        this.author = author;
    }

    public abstract void display();
}

class MyBook : Book
{
    private int price;

    public MyBook(string title, string author, int price)
        : base(title, author)
    {
        this.price = price;
    }

    public override void display()
    {
        Console.WriteLine("Title: " + title);
        Console.WriteLine("Author: " + author);
        Console.WriteLine("Price: " + price);
    }
}

class Solution
{
    static void Main(string[] args)
    {
        string title = Console.ReadLine();
        string author = Console.ReadLine();
        int price = Convert.ToInt32(Console.ReadLine());

        Book book = new MyBook(title, author, price);
        book.display();
    }
}